interpreter/spec/eval.ml

```diff
@@ -398,8 +453,11 @@ let check_limits actual expected at =
 let add_import (ext : extern) (im : import) (inst : instance) : instance =
   let {ikind; _} = im.it in
   match ext, ikind.it with
-  | ExternalFunc clos, FuncImport x when func_type_of clos = type_ inst x ->
-    {inst with funcs = clos :: inst.funcs}
+  | ExternalFunc clos, FuncImport x
+    when func_type_of clos = match type_ inst x with
```

Comment:
This occurs repeatedly. How about a helper `func_type : inst -> var -> func_type`?